app/main.py

```diff
@@ -2,0 +1,34 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return "{"\
+               f"Name: {self.name}, " \
+               f"Health: {self.health}," \
+               f" Hidden: {self.hidden}"\
+               "}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(animal: Herbivore) -> None:
+        if not animal.hidden and isinstance(animal, Herbivore):
```

Comment:
The `isinstance(animal, Herbivore)` check is redundant here because the method signature already specifies that `animal` should be a `Herbivore`. You can safely remove this check.